src/entity.unit.ts

```diff
@@ -136,42 +151,109 @@ describe("Entity", () => {
+				if (prop.name === "Budget" && value === "BUDGET_1")
+					return Promise.resolve(Budget1);
+			});
+			const movie = new Types.Movie({ ...Alien, Budget: "BUDGET_1" });
+			const expected = { ...Alien, Budget: Budget1 };
+			expect(movie.serialize()).not.toEqual(expected);
+			await movie.initialized;
+			expect(movie.serialize()).toEqual(expected);
+		});
 	});
 
-	describe("set", () => {
+	describe("update", () => {
 		it("can be used to update an entity", () => {
 			const movie = new Types.Movie();
-			movie.set(Alien);
+			movie.update(Alien);
 			expect(movie.serialize()).toEqual(Alien);
 		});
 
 		it("cannot be used to set calculated properties", () => {
 			const person = new Types.Person({ FirstName: "John", LastName: "Doe" });
-			person.set({ FullName: "Full Name" });
+			person.update({ FullName: "Full Name" });
 			expect(person.FullName).toBe("John Doe");
 		});
 
 		it("cannot be used to set constant properties", () => {
 			const person = new Types.Person();
-			person.set({ Species: "Homo erectus" });
+			person.update({ Species: "Homo erectus" });
 			expect(person.Species).toBe("Homo sapiens");
 		});
 
 		it("cannot be used to set a value deserialized to undefined", () => {
 			const person = new Types.Person(Alien.Director);
 			const movieBeforeSet = person.Movie;
-			jest.spyOn(person.serializer, "deserialize").mockImplementation(()=>undefined);
-			person.set({ Movie: Alien });
+			jest.spyOn(person.serializer, "deserialize").mockImplementation(() => undefined);
+			person.update({ Movie: Alien });
 			expect(person.Movie).toBe(movieBeforeSet);
 		});
 
 		it("cannot be used to set a value deserialized to undefined in a list", () => {
 			const movie = new Types.Movie({ Cast: [{ FirstName: "John", LastName: "Doe" }] });
 			const movieBeforeSet = movie.Cast;
-			jest.spyOn(movie.serializer, "deserialize").mockImplementation(()=>undefined);
-			movie.set({ Cast: [{ FirstName: "Ridley", LastName: "Scott" }, { FirstName: "John", LastName: "Doe" }] });
+			jest.spyOn(movie.serializer, "deserialize").mockImplementation(() => undefined);
+			movie.update({ Cast: [{ FirstName: "Ridley", LastName: "Scott" }, { FirstName: "John", LastName: "Doe" }] });
 			expect(movie.Cast).toBe(movieBeforeSet);
 		});
+
+		it("allows waiting for asynchronous values to be set", async () => {
+			const Budget1 = { LineItems: [{ Label: "L1", Cost: 1000000 }] };
+
+			model.serializer.registerValueResolver((entity, prop, value) => {
+				if (prop.name === "Budget" && value === "BUDGET_1")
+					return Promise.resolve(Budget1);
+			});
+
+			const movie = new Types.Movie(Alien);
+
+			const updateTask = movie.update({ Title: "Alien 2", Budget: "BUDGET_1" });
+
+			expect(movie.Title).toBe("Alien 2");
+			expect(movie.Budget).toBeNull();
+
+			await updateTask;
+
+			expect(movie.Budget.serialize()).toEqual(Budget1);
+		});
+
+		it("should support circular async value resolution", async () => {
```

Comment:
This test moved from `type.unit.ts`, since it was previously exercising the "update" aspect of `Type.create`, which is no longer a thing.